arch/riscv/Kconfig

```
@@ -166,7 +166,9 @@ config PMP_POWER_OF_TWO_ALIGNMENT
 endif #RISCV_PMP
 
 config PMP_STACK_GUARD
-	def_bool y
+	bool
+	default n if !MULTITHREADING
+	default y
```

minor nit, it would be sufficient to move the condition to `default y` as `n` is default in Kconfig when not stated, that is:
```suggestion
	default y if MULTITHREADING
```

imo it also makes it clearer that pmp stack guard enabling is depending on multi threading.